backend/src/contaxy/managers/deployment/docker_utils.py

```diff
@@ -211,24 +211,63 @@ def handle_network(
                 Labels.PROJECT_NAME.value: project_id,
             },
         )
+    connect_to_network(get_this_container(client), network)
+    return network
+
 
+def connect_to_network(
+    container: docker.models.containers.Container,
+    network: docker.models.networks.Network,
+) -> None:
     # Note: connecting the backend container to the network will cause a connection drop on Macs, so the first service deploy call will be shown as failed, even though it succeeded. On Linux, this problem does not seem to exist
-    backend_container = get_this_container(client)
-    if backend_container:
-        is_backend_connected_to_network = backend_container.attrs["NetworkSettings"][
+    if container:
+        is_backend_connected_to_network = container.attrs["NetworkSettings"][
             "Networks"
-        ].get(network_name, False)
+        ].get(network.name, False)
         if not is_backend_connected_to_network:
             try:
-                network.connect(backend_container)
+                network.connect(container)
             except docker.errors.APIError:
                 # Remove the network again as it is not connected to any service.
                 network.remove()
                 raise RuntimeError(
-                    f"Could not connect the backend container to the network {network_name}"
+                    f"Could not connect the {container.name} to the network {network.name}"
                 )
 
-    return network
+
+def reconnect_to_all_networks(
+    client: docker.client,
+) -> None:
+    """Connects the backend container to all networks that belong to the installation.
+
+    A reconnect is required after the container was recreated (e.g. to update the image) and is therefore
+    no longer part of all the networks.
+    :param client: Docker client
+    """
+    networks = get_backend_networks(client)
+    for network in networks:
+        try:
+            connect_to_network(get_this_container(client), network)
+        except RuntimeError:
```

Comment:
I think we would need to use the thrown error here to know which network was not connected, right? Something like

```python
except RuntimeError as e:
  logger.exception(e)
```

Reply:
As far as I know the python Logger.exception automatically adds the stack trace of the caught exception:
https://docs.python.org/3/library/logging.html#logging.Logger.exception
I guess we are using loguru here. They have an example that shows how the exception stack trace if automatically added to the log output:
https://github.com/Delgan/loguru#fully-descriptive-exceptions